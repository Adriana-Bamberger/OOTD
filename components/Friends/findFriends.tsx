import React, { useState } from 'react'
import { View, Text, TextInput, FlatList, StyleSheet } from 'react-native'

// Dummy data for friends (replace with actual data from your database)
const friendsData = [
  { id: '1', name: 'Alice' },
  { id: '2', name: 'Bob' },
  { id: '3', name: 'Charlie' },
  { id: '4', name: 'David' },
  { id: '5', name: 'Emma' },
  { id: '6', name: 'Frank' },
  { id: '7', name: 'Grace' },
  { id: '8', name: 'Hannah' },
  { id: '9', name: 'Isaac' },
  { id: '10', name: 'Jack' },
  { id: '11', name: 'Felix' },
  { id: '12', name: 'Adriana' },
  { id: '13', name: 'Val' },
  { id: '14', name: 'Bevan' },
  { id: '15', name: 'Vin' },
  // Add more friends as needed
]

const FindFriends = () => {
  const [searchQuery, setSearchQuery] = useState('')
  const [filteredFriends, setFilteredFriends] = useState(friendsData)

  // Function to handle search input change
  const handleSearch = (text: string) => {
    setSearchQuery(text)
    const filteredData = friendsData.filter((friend) =>
      friend.name.toLowerCase().includes(text.toLowerCase())
    )
    setFilteredFriends(filteredData)
  }

  // Render each friend item
  const renderFriend = ({ item }: { item: { id: string; name: string } }) => (
    <View style={styles.friendItem}>
      <Text>{item.name}</Text>
    </View>
  )

  return (
    <View style={styles.container}>
      <TextInput
        style={styles.searchInput}
        placeholder="Search friends..."
        value={searchQuery}
        onChangeText={handleSearch}
        autoCorrect={false}
        autoFocus
      />
      <FlatList
        data={filteredFriends}
        renderItem={renderFriend}
        keyExtractor={(item) => item.id}
        style={styles.friendList}
      />
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    paddingHorizontal: 16,
    paddingTop: 16,
  },
  searchInput: {
    height: 40,
    borderColor: '#ccc',
    borderWidth: 1,
    borderRadius: 8,
    paddingHorizontal: 16,
    marginBottom: 16,
  },
  friendList: {
    flex: 1,
    width: '100%',
  },
  friendItem: {
    paddingVertical: 12,
    borderBottomWidth: 1,
    borderBottomColor: '#ddd',
  },
})

export default FindFriends
