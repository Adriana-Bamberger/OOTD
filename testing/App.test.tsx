//This page makes basic snapshot tests for our main pages. - Adriana
// TODO: come back to make the one for Profile page, without it beign attached to our firebase DB yet it's really hard to test.
import React from 'react'
import renderer from 'react-test-renderer'
import { describe, expect, test } from '@jest/globals'
import Friends from '@/app/(tabs)/friends'

describe('Friends component', () => {
  it('renders the component correctly', () => {
    const tree = renderer.create(<Friends />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

import Feed from '@/app/(tabs)/feed'

describe('Feed component', () => {
  it('renders the component with dummy data correctly', () => {
    const tree = renderer.create(<Feed />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})

import App from '@/app/(tabs)/carousel'

describe('Carousel component', () => {
  it('renders the component with empty data', () => {
    const tree = renderer.create(<App />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
