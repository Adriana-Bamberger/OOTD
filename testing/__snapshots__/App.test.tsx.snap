// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel component renders the component with empty data 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      {
        "alignItems": "center",
      }
    }
  >
    <View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "marginTop": 20,
          }
        }
      >
        <Text
          style={
            {
              "fontFamily": "TYPOGRAPH-PRO-Semi-Bold",
              "fontSize": 50,
              "padding": 5,
            }
          }
        >
          OOTD.
        </Text>
      </View>
      <View
        style={
          {
            "flexDirection": "row",
            "justifyContent": "space-around",
            "marginBottom": 10,
            "marginTop": 10,
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "minWidth": "40%",
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 20,
                "fontWeight": "bold",
                "marginBottom": 10,
                "textAlign": "center",
              }
            }
          >
            Headwear
          </Text>
          <View
            style={
              {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "justifyContent": "center",
                }
              }
            >
              <Image
                source={
                  {
                    "uri": "https://media.tenor.com/images/3704f2b9b8b66a5747116f436a5e6aba/tenor.gif",
                  }
                }
                style={
                  {
                    "borderRadius": 20,
                    "height": 150,
                    "width": 150,
                  }
                }
              />
            </View>
          </View>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "minWidth": "40%",
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 20,
                "fontWeight": "bold",
                "marginBottom": 10,
                "textAlign": "center",
              }
            }
          >
            Accessories
          </Text>
          <View
            style={
              {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "justifyContent": "center",
                }
              }
            >
              <Image
                source={
                  {
                    "uri": "https://media.tenor.com/images/3704f2b9b8b66a5747116f436a5e6aba/tenor.gif",
                  }
                }
                style={
                  {
                    "borderRadius": 20,
                    "height": 150,
                    "width": 150,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "flexDirection": "row",
            "justifyContent": "space-around",
            "marginBottom": 10,
            "marginTop": 10,
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "minWidth": "40%",
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 20,
                "fontWeight": "bold",
                "marginBottom": 10,
                "textAlign": "center",
              }
            }
          >
            Outerwear
          </Text>
          <View
            style={
              {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "justifyContent": "center",
                }
              }
            >
              <Image
                source={
                  {
                    "uri": "https://media.tenor.com/images/3704f2b9b8b66a5747116f436a5e6aba/tenor.gif",
                  }
                }
                style={
                  {
                    "borderRadius": 20,
                    "height": 150,
                    "width": 150,
                  }
                }
              />
            </View>
          </View>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "minWidth": "40%",
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 20,
                "fontWeight": "bold",
                "marginBottom": 10,
                "textAlign": "center",
              }
            }
          >
            Tops
          </Text>
          <View
            style={
              {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "justifyContent": "center",
                }
              }
            >
              <Image
                source={
                  {
                    "uri": "https://media.tenor.com/images/3704f2b9b8b66a5747116f436a5e6aba/tenor.gif",
                  }
                }
                style={
                  {
                    "borderRadius": 20,
                    "height": 150,
                    "width": 150,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "flexDirection": "row",
            "justifyContent": "space-around",
            "marginBottom": 10,
            "marginTop": 10,
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "minWidth": "40%",
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 20,
                "fontWeight": "bold",
                "marginBottom": 10,
                "textAlign": "center",
              }
            }
          >
            Bottoms
          </Text>
          <View
            style={
              {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "justifyContent": "center",
                }
              }
            >
              <Image
                source={
                  {
                    "uri": "https://media.tenor.com/images/3704f2b9b8b66a5747116f436a5e6aba/tenor.gif",
                  }
                }
                style={
                  {
                    "borderRadius": 20,
                    "height": 150,
                    "width": 150,
                  }
                }
              />
            </View>
          </View>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "minWidth": "40%",
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 20,
                "fontWeight": "bold",
                "marginBottom": 10,
                "textAlign": "center",
              }
            }
          >
            Footwear
          </Text>
          <View
            style={
              {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "justifyContent": "center",
                }
              }
            >
              <Image
                source={
                  {
                    "uri": "https://media.tenor.com/images/3704f2b9b8b66a5747116f436a5e6aba/tenor.gif",
                  }
                }
                style={
                  {
                    "borderRadius": 20,
                    "height": 150,
                    "width": 150,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          {
            "flexDirection": "row",
            "justifyContent": "space-around",
            "marginBottom": 20,
            "paddingHorizontal": 20,
            "width": "100%",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "black",
              "borderRadius": 8,
              "opacity": 1,
              "paddingHorizontal": 20,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "white",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Shuffle
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "black",
              "borderRadius": 8,
              "opacity": 1,
              "paddingHorizontal": 20,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "white",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Save
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;

exports[`Feed component renders the component with dummy data correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "padding": 20,
    }
  }
  data={
    [
      {
        "avatar": 1,
        "caption": "erydey cri",
        "id": "1",
        "image": 1,
        "likes": 10,
        "username": "mintswagger42069",
      },
      {
        "avatar": 1,
        "caption": "i cry i cry",
        "id": "2",
        "image": 1,
        "likes": 5,
        "username": "bananastick100",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={0.0001}
  stickyHeaderIndices={[]}
  viewabilityConfigCallbackPairs={[]}
>
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          {
            "backgroundColor": "#f8f8f8",
            "borderRadius": 10,
            "marginBottom": 20,
            "padding": 15,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "marginBottom": 10,
            }
          }
        >
          <Image
            source={1}
            style={
              {
                "borderRadius": 25,
                "height": 50,
                "marginRight": 10,
                "width": 50,
              }
            }
          />
          <Text
            style={
              {
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            mintswagger42069
          </Text>
        </View>
        <Image
          source={1}
          style={
            {
              "borderRadius": 10,
              "height": 300,
              "width": "100%",
            }
          }
        />
        <Text
          style={
            {
              "fontSize": 14,
              "marginBottom": 5,
            }
          }
        >
          erydey cri
        </Text>
        <Text
          style={
            {
              "fontSize": 14,
              "fontWeight": "bold",
              "marginBottom": 10,
            }
          }
        >
          10
           likes
        </Text>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#3498db",
              "borderRadius": 5,
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "#fff",
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
            Like
          </Text>
        </View>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          {
            "backgroundColor": "#f8f8f8",
            "borderRadius": 10,
            "marginBottom": 20,
            "padding": 15,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "marginBottom": 10,
            }
          }
        >
          <Image
            source={1}
            style={
              {
                "borderRadius": 25,
                "height": 50,
                "marginRight": 10,
                "width": 50,
              }
            }
          />
          <Text
            style={
              {
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            bananastick100
          </Text>
        </View>
        <Image
          source={1}
          style={
            {
              "borderRadius": 10,
              "height": 300,
              "width": "100%",
            }
          }
        />
        <Text
          style={
            {
              "fontSize": 14,
              "marginBottom": 5,
            }
          }
        >
          i cry i cry
        </Text>
        <Text
          style={
            {
              "fontSize": 14,
              "fontWeight": "bold",
              "marginBottom": 10,
            }
          }
        >
          5
           likes
        </Text>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#3498db",
              "borderRadius": 5,
              "opacity": 1,
              "padding": 10,
            }
          }
        >
          <Text
            style={
              {
                "color": "#fff",
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
            Like
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`Friends component renders the component correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "flex": 1,
      "paddingHorizontal": 16,
      "paddingTop": 16,
    }
  }
>
  <TextInput
    autoCorrect={false}
    autoFocus={true}
    onChangeText={[Function]}
    placeholder="Search friends..."
    style={
      {
        "borderColor": "#ccc",
        "borderRadius": 8,
        "borderWidth": 1,
        "height": 40,
        "marginBottom": 16,
        "paddingHorizontal": 16,
      }
    }
    value=""
  />
  <RCTScrollView
    data={
      [
        {
          "id": "1",
          "name": "Alice",
        },
        {
          "id": "2",
          "name": "Bob",
        },
        {
          "id": "3",
          "name": "Charlie",
        },
        {
          "id": "4",
          "name": "David",
        },
        {
          "id": "5",
          "name": "Emma",
        },
        {
          "id": "6",
          "name": "Frank",
        },
        {
          "id": "7",
          "name": "Grace",
        },
        {
          "id": "8",
          "name": "Hannah",
        },
        {
          "id": "9",
          "name": "Isaac",
        },
        {
          "id": "10",
          "name": "Jack",
        },
        {
          "id": "11",
          "name": "Felix",
        },
        {
          "id": "12",
          "name": "Adriana",
        },
        {
          "id": "13",
          "name": "Val",
        },
        {
          "id": "14",
          "name": "Bevan",
        },
        {
          "id": "15",
          "name": "Vin",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    style={
      {
        "flex": 1,
        "width": "100%",
      }
    }
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "borderBottomColor": "#ddd",
              "borderBottomWidth": 1,
              "paddingVertical": 12,
            }
          }
        >
          <Text>
            Alice
          </Text>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "borderBottomColor": "#ddd",
              "borderBottomWidth": 1,
              "paddingVertical": 12,
            }
          }
        >
          <Text>
            Bob
          </Text>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "borderBottomColor": "#ddd",
              "borderBottomWidth": 1,
              "paddingVertical": 12,
            }
          }
        >
          <Text>
            Charlie
          </Text>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "borderBottomColor": "#ddd",
              "borderBottomWidth": 1,
              "paddingVertical": 12,
            }
          }
        >
          <Text>
            David
          </Text>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "borderBottomColor": "#ddd",
              "borderBottomWidth": 1,
              "paddingVertical": 12,
            }
          }
        >
          <Text>
            Emma
          </Text>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "borderBottomColor": "#ddd",
              "borderBottomWidth": 1,
              "paddingVertical": 12,
            }
          }
        >
          <Text>
            Frank
          </Text>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "borderBottomColor": "#ddd",
              "borderBottomWidth": 1,
              "paddingVertical": 12,
            }
          }
        >
          <Text>
            Grace
          </Text>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "borderBottomColor": "#ddd",
              "borderBottomWidth": 1,
              "paddingVertical": 12,
            }
          }
        >
          <Text>
            Hannah
          </Text>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "borderBottomColor": "#ddd",
              "borderBottomWidth": 1,
              "paddingVertical": 12,
            }
          }
        >
          <Text>
            Isaac
          </Text>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "borderBottomColor": "#ddd",
              "borderBottomWidth": 1,
              "paddingVertical": 12,
            }
          }
        >
          <Text>
            Jack
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "height": 0,
          }
        }
      />
    </View>
  </RCTScrollView>
</View>
`;
